improver_tests/utilities/test_OccurrenceWithinVicinity.py

```diff
@@ -257,16 +203,9 @@ def test_with_multiple_realizations_and_times(self):
                 ],
             ]
         )
-        data = np.zeros((2, 2, 4, 4))
-        data[0, 0, 2, 1] = 1.0
-        data[1, 1, 1, 3] = 1.0
-        cube = set_up_cube(
-            data,
-            "lwe_precipitation_rate",
-            "m s-1",
-            timesteps=np.array([402192.5, 402195.5]),
-            realizations=np.array([0, 1]),
-        )
+        cube = add_coordinate(self.cube, self.timesteps, "time", order=[1, 0, 2, 3])
```

Comment:
We should add some units to the time coordinate, and ensure it has a suitable data type.
```suggestion
        cube = add_coordinate(
            self.cube,
            self.timesteps,
            "time",
            order=[1, 0, 2, 3],
            coord_units="seconds since 1970-01-01 00:00:00",
            dtype=np.int64,
        )
```

Reply:
Ah I missed this!  Can I add to Ben's comments here: when using this function to add a time dimension, it is much preferred to use the `is_datetime` argument, for example:
```
cube = add_coordinate(
    self.cube, [datetime(2017, 11, 9, 12), datetime(2017, 11, 9, 15)], "time",
    order=[1, 0, 2, 3], is_datetime=True
)
```
It was designed this way to make the tests more human-readable, and so that if we do ever want to change the handling of time coordinates (units or datatypes), we only have to update that once in the `set_up_test_cubes` module.

Please could you update that wherever it's used, rather than setting the time point values as integers?

Thanks.